### ISSUE TITLE
Panic "fatal error: runtime: mcall called on m->g0 stack"

### ISSUE THREAD

**Issue by lxndrcx** [syncthing/syncthing#9509]
### What happened?

Synthing runs fine for a while, then panics. It has successfully synced previously. I tried reinstalling and creating a new config, the crash still occurs.
Redacted panic log included.
[panic-20240410-164317.log](https://github.com/syncthing/syncthing/files/14927469/panic-20240410-164317.log)

### Syncthing version

 v1.27.6, Windows (64-bit Intel/AMD) "Gold Grasshopper" 

### Platform & operating system

Windows Server 2019 VM x64

### Relevant log output

```shell
[start] 2024/04/10 16:42:27 INFO: syncthing v1.27.6 "Gold Grasshopper" (go1.22.2 windows-amd64) builder@github.syncthing.net 2024-04-08 09:14:27 UTC
[62NOC] 2024/04/10 16:42:27 INFO: My ID: ***
[62NOC] 2024/04/10 16:42:28 INFO: Single thread SHA256 performance is 1956 MB/s using crypto/sha256 (1939 MB/s using minio/sha256-simd).
[62NOC] 2024/04/10 16:42:28 INFO: Hashing performance is 1146.62 MB/s
[62NOC] 2024/04/10 16:42:28 INFO: Overall send rate is unlimited, receive rate limit is 25000 KiB/s
[62NOC] 2024/04/10 16:42:28 INFO: Rate limits do not apply to LAN connections
[62NOC] 2024/04/10 16:42:28 INFO: Using discovery mechanism: global discovery server https://discovery.syncthing.net/v2/?noannounce&id=LYXKCHX-VI3NYZR-ALCJBHF-WMZYSPK-QG6QJA3-MPFYMSO-U56GTUK-NA2MIAW
[62NOC] 2024/04/10 16:42:28 INFO: Ready to synchronize "Default Folder" (default) (sendreceive)
[62NOC] 2024/04/10 16:42:28 INFO: Ready to synchronize "To" (dz79c-qrtru) (receiveonly)
[62NOC] 2024/04/10 16:42:28 INFO: Using discovery mechanism: global discovery server https://discovery-v4.syncthing.net/v2/?nolookup&id=LYXKCHX-VI3NYZR-ALCJBHF-WMZYSPK-QG6QJA3-MPFYMSO-U56GTUK-NA2MIAW
...
[62NOC] 2024/04/10 16:42:28 INFO: Using discovery mechanism: global discovery server https://discovery-v6.syncthing.net/v2/?nolookup&id=LYXKCHX-VI3NYZR-ALCJBHF-WMZYSPK-QG6QJA3-MPFYMSO-U56GTUK-NA2MIAW
[62NOC] 2024/04/10 16:42:28 INFO: Using discovery mechanism: IPv4 local broadcast discovery on port 21027
[62NOC] 2024/04/10 16:42:28 INFO: Using discovery mechanism: IPv6 local multicast discovery on address [ff12::8384]:21027
[62NOC] 2024/04/10 16:42:28 INFO: Completed initial scan of receiveonly folder "To" (dz79c-qrtru)
[62NOC] 2024/04/10 16:42:29 INFO: GUI and API listening on 127.0.0.1:8384
[62NOC] 2024/04/10 16:42:29 INFO: Access the GUI via the following URL: http://127.0.0.1:8384/
[62NOC] 2024/04/10 16:42:29 INFO: My name is "WIN-***"
[62NOC] 2024/04/10 16:42:29 INFO: Device *** is "sender" at [dynamic]
[62NOC] 2024/04/10 16:42:29 INFO: Completed initial scan of sendreceive folder "Default Folder" (default)
[62NOC] 2024/04/10 16:42:29 INFO: QUIC listener ([::]:9099) starting
[62NOC] 2024/04/10 16:43:06 INFO: Device 6PBSMA6 client is "syncthing v1.27.6" named "sender" at [::]:9099-***:7454/quic-server/TLS1.3-TLS_AES_128_GCM_SHA256/WAN-P30-5U4Q69AOON7G8UG8448ESIS91O
Panic at 2024-04-10T16:43:19+12:00
fatal error: runtime: mcall called on m->g0 stack

runtime stack:
runtime.throw({0x1e0690a?, 0x10?})
	runtime/panic.go:1023 +0x65 fp=0xc000e730f8 sp=0xc000e730c8 pc=0x12000a5
runtime.badmcall(0xc000e732c8?)
	runtime/proc.go:507 +0x1f fp=0xc000e73118 sp=0xc000e730f8 pc=0x120347f
runtime.badmcall(0x1201539)
	<autogenerated>:1 +0x25 fp=0xc000e73130 sp=0xc000e73118 pc=0x123e645
runtime: g 0: unexpected return pc for runtime.badmcall called from 0xc000e73148
stack: frame={sp:0xc000e73118, fp:0xc000e73130} stack=[0x8770404000,0x87705ff9a0)


[log truncated, see attached file]
```

**Comment by calmh** [comment]
The crash is in the Go runtime, which points to either a bug in Go (somewhat unlikely, since this appears to happen to you regularly but mostly only you) or a problem with your machine. My gut feeling says the latter... Hardware problem, virtualisation problem, something like that?

**Comment by lxndrcx** [comment]
You're probably right. I just wanted to report it in case there is
something I didn't know about, and for anyone else who gets a similar
error.
I suspect the VM was not set up well, as I wasn't responsible for it.
Thanks for your response.

**Comment by RichieSams** [comment]
Question: What hardware was your VM running on? AMD Threadripper / EPYC by chance?
I'm seeing this error frequently on our VM fleet, but *only* on AMD Threadripper machines. I saw this golang bug, but it was closed when the user stopped seeing it. I have updated past golang 1.19.3, but still see the issue.

https://github.com/golang/go/issues/56774

(To @calmh this isn't related to synchthing. Just trying to gather more information where I can find it. Apologies)
